tests/yelp.pm

```diff
@@ -0,0 +1,40 @@
+# vi: set shiftwidth=4 tabstop=4 expandtab:
+use base 'installedtest';
+use strict;
+use testapi;
+use utils;
+
+sub run {
+    my $self = shift;
+
+    check_desktop_clean();
+
+    # Run Yelp; it should show our customised main page.
+    type_very_safely("yelp\n");
+    assert_screen('yelp_main_page', 10);
+
+    send_key_combo('alt', 'f4');
+
+    # Search for a help topic, and results should be shown by a search provider.
+    type_very_safely('add user');
+    assert_screen('yelp_search_provider_results', 10);
```

Comment:
Might be nice to match one of the results, click on it, and check it works, but that could be something for the future.